### ISSUE TITLE
Problem with new code for location specific media likes

### ISSUE THREAD

**Issue by Bigemm** [instabot-py/instabot.py#594]
![error](https://user-images.githubusercontent.com/28912732/27061173-89e9f4fe-5014-11e7-8441-cc14c1eb10d5.jpg)

bot claims to be unable to decode as show in picture, additionally, normal liking mechanism has stopped working after months of carrying out task (even though I am not blocked on instagram). 

I am running this over pythonanywhere dot com which runs it on python 2.7 - could this have anything to do with it?

**Comment by badbotnba** [comment]
i am also having this problem

**Comment by ThomasMW** [comment]
I am retrieving location feeds just fine. However, do you see that 400 status code when you try to like a specific media. It means Instagram is percieving your requests as spam, a lot of users running their bot on PythonAnywhere are having this problem.

I would advise taking the bot off PythonAnywhere and running it from another platform, this seems to help solve the problem.

**Comment by Bigemm** [comment]
@ThomasMW  are you running it from python 3 (desktop) - I will try to run it from there and report back. Additionally, have you had any issues with the liking mechanism? or is your bot running completely fine?

**Comment by ThomasMW** [comment]
Running the bot from another platform like your own PC helps the problem because you're behind a different IP. Because more people started using PythonAnywhere Instagram started to receive more requests from the same IPs. When this passes the threshold they have in place the IP get's softbanned which means certain requests like liking and commenting don't work anymore.

**Comment by drovid** [comment]
this is instagram's IP ban wave i strongly urge everyone to switch IP adresses and run their bots in lite mode (about 1/10th of regular rate)

**Comment by Bigemm** [comment]
I see. If this is really an IP problem, it's only a matter of time before Instagram catches on to IP on other platforms I use and same result will inevitably occur especially since I will be running multiple accounts from one machine. Would you advise applying proxy for each account and running lite? 

Going to kill script for a week or 2 then pickup after again.

**Comment by dcanones** [comment]
I am running some bots and have 10 private proxies. Having like 2-3 bots per proxy, Instagram is starting to ban likes (400 error) in some of the bots, so the threshold per ip must be extremely low now. It seems like they are fighting hard again bots.

**Comment by alexfromvl** [comment]
Here's my code modification, which works fine.
`
    def get_media_id_by_location(self, location):
        """ Get media ID set, by your location number """

        if (self.login_status):
            log_string = "Get media id by location: %s" % (location)
            self.write_log(log_string)
            if self.login_status == 1:
                url_location = '%s%s%s' % (self.url_location, location, '/')
                try:
                    r = self.s.get(url_location)
                    text = r.text

                    finder_text_start = ('<script type="text/javascript">'
                                         'window._sharedData = ')
                    finder_text_start_len = len(finder_text_start)-1
                    finder_text_end = ';</script>'

                    all_data_start = text.find(finder_text_start)
                    all_data_end = text.find(finder_text_end, all_data_start + 1)
                    json_str = text[(all_data_start + finder_text_start_len + 1) \
                                   : all_data_end]
                    all_data = json.loads(json_str)

                    self.media_by_tag = list(all_data['entry_data']['LocationsPage'][0]\
                                            ['location']['media']['nodes'])
                except:
                    self.media_by_tag = []
                    self.write_log("Except on get_media!")
                    time.sleep(60)
            else:
                return 0
`

**Comment by Bigemm** [comment]
@alexfromvl thanks for that, haven't had a chance to test yet but will do and let you know if it works for me.

**Comment by lmpan** [comment]
@alexfromvl Hi Alex, how does this function work?  I included this function into new_auto_mode(). But how can I get the location id. Any hint are greatly appreciated. Thank you very much!

**Comment by alexfromvl** [comment]
@lmpan Hi. I did not do an additional handler to find a place by name. Now I'm using the location numbers. I take the number id location on a direct request through the search for an instagram.

**Comment by ugurozturk** [comment]
Major changes have been made to Instabot.py. If this is still an issue, please open a new ticket.
